datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -372,6 +377,40 @@ pub fn date_part(args: &[ColumnarValue]) -> Result<ColumnarValue> {
     })
 }
 
+fn extract_epoch<T>(array: &PrimitiveArray<T>) -> Result<Int32Array>
+where
+    T: ArrowTemporalType + ArrowNumericType,
+    i64: std::convert::From<T::Native>,
+{
+    let mut b = Int32Builder::new(array.len());
+    match array.data_type() {
+        &DataType::Timestamp(ref unit, _) => {
+            for i in 0..array.len() {
+                if array.is_null(i) {
+                    b.append_null()?;
+                } else {
+                    let scale = match unit {
```

Comment:
I would recommend using the `chrono` functions for this, such as done here:

https://sourcegraph.com/github.com/apache/arrow-rs/-/blob/arrow/src/compute/kernels/temporal.rs?L214-240&subtree=true